include/algorithm/merge_sort.h

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
Rename file to `mergesort.h`

Follow the code style specified in [wiki](https://github.com/ParadoxZero/libgeneric/wiki/Code-guide)